src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> lastNode = tail;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        tail = newNode;
+        if (lastNode == null) {
+            head = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> currentNode = node(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        Node<T> previousNode = currentNode.prev;
+        if (previousNode == null) {
+            head = newNode;
+        } else {
+            previousNode.next = newNode;
+        }
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToChange = node(index);
+        T oldItem = nodeToChange.item;
+        nodeToChange.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldNode = node(index);
+        return unlink(oldNode);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((currentNode.item == object)
+                    || (object != null && object.equals(currentNode.item))) {
+                remove(i);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("No such index in data");
+        }
+    }
+
+    private Node<T> node(int index) {
```

Comment:
```suggestion
    private Node<T> findNode(int index) {
```
let's use verbs for method names